llvm/lib/CodeGen/GlobalISel/CallLowering.cpp

```diff
@@ -764,19 +765,28 @@ bool CallLowering::handleAssignments(ValueHandler &Handler,
       // If we can't directly assign the register, we need one or more
       // intermediate values.
       Args[i].Regs.resize(NumParts);
-
-      // For each split register, create and assign a vreg that will store
-      // the incoming component of the larger value. These will later be
-      // merged to form the final vreg.
-      for (unsigned Part = 0; Part < NumParts; ++Part)
-        Args[i].Regs[Part] = MRI.createGenericVirtualRegister(NewLLT);
+    
+      // When we have indirect parameter passing we are receiving a pointer,
+      // that points to the actual value. In that case we need a pointer.
+      if (VA.getLocInfo() == CCValAssign::Indirect &&
+          Args[i].Flags[0].isSplit()) {
+        if (Handler.isIncomingArgumentHandler())
```

Comment:
Why don't we need to handle the `!Handler.isIncomingArgumentHandler()` case? We handle both cases for non-indirect parameter passing.

Reply:
In the case when we are receiving the call (so we are the callee) We have to create these "temporaries" where we can copy the pointer, so we can call `G_LOAD` on it.
The callee point of view:
```llvm
  %2:_(p0) = COPY $x10 ; here we use the tempoaray regsiter created by the highlighted code to copy the incoming value here
  %0:_(s128) = G_LOAD %2:_(p0) :: (load (s128), align 1) 
```

The caller point of view:
```llvm
  %1:_(s128) = G_CONSTANT i128 1
  %3:_(p0) = G_FRAME_INDEX %stack.0
  G_STORE %2:_(s128), %4:_(p0) :: (store (s128), align 8) ; Here we already have a pointer received from the frame index creation. And the value to pass is already existing, so that temporary is not needed.
  $x11 = COPY %4:_(p0)
```

Reply:
Yes I agree with you. I should put some more explanation it the code comments too. Not just in this place but other places too.

Reply:
I have added some more comments and examples to better explain what is happening when handling indirect parameter passing.